scripts/pl-install.sh

```diff
@@ -78,9 +78,6 @@ arch="$(uname -m)"
 curl -LO https://github.com/conda-forge/miniforge/releases/download/24.3.0-0/Miniforge3-Linux-${arch}.sh
 bash Miniforge3-Linux-${arch}.sh -b -p /usr/local -f
 
-echo "installing Python packages..."
-python3 -m pip install --no-cache-dir -r /PrairieLearn/requirements.txt
```

Comment:
We lost the `--no-cache-dir` here, which is important for minimizing the size of the resulting image. You can see the impact here for yourself:

<img width="388" alt="Screenshot 2025-07-02 at 10 29 20" src="https://github.com/user-attachments/assets/3896a85f-c4d3-45ad-9425-662f627772de" />

I'm assuming this increase is mostly due to the cache, there could be other factors involved here too. They'll need to be addressed before this can be merged.

Reply:
This makes me think that installing via the makefile isn't the best idea anymore, since the command would change.

Reply:
Proposal: setup venv using makefile, install packages via a command?

Reply:
That's not too crazy; we run `yarn install` directly in the `Dockerfile` already! Though if we wanted to continue delegating to the makefile, we could [`pip cache purge`](https://pip.pypa.io/en/stable/cli/pip_cache/) after install (just like we `yarn cache clean` after installing Yarn deps).

https://pip.pypa.io/en/stable/topics/configuration/#environment-variables would imply that we can also do something like `PIP_NO_CACHE_DIR=1` if we wanted to influence pip's behavior without changing the command in the Makefile itself.